### PR TITLE
Add diagonal move blocks to K-1 playlab

### PR DESCRIPTION
**Pull request by balderdash**
image

The move distance is longer than normal by a factor of sqrt(2), so that a move north followed by a move west takes you to the same spot as a single move northwest.

### CODE REVIEW

File: apps/src/studio/blocks.js [hunk 1]
```diff
@@ -970,9 +994,13 @@ exports.install = function (blockly, blockInstallOptions) {
...
       return function () {
         var sprite = this.getTitleValue('SPRITE') || '0';
-        var direction = directionConfig.studioValue;
+        var direction = directionConfig.studioValue.toString();
         var methodName = isEventMove ? 'move' : 'moveDistance';
-        var distance = this.getTitleValue('DISTANCE') || SimpleMove.DEFAULT_MOVE_DISTANCE;
+        var defaultDistance =
+          constants.CardinalDirections.includes(directionConfig.studioValue) ?
+          SimpleMove.DEFAULT_MOVE_DISTANCE :
+          SimpleMove.DEFAULT_MOVE_DISTANCE * Math.sqrt(2);
```
Comment (Hamms) [inline apps/src/studio/blocks.js:1005]: Might be worth including what you mentioned in the PR description here as a comment

File: apps/src/studio/studio.js [hunk 2]
```diff
@@ -611,40 +612,28 @@ var delegate = function (scope, func, data) {
...
-        case Direction.SOUTH:
-          scaleFactor = yAxis ? 1 : 0;
-          break;
-        case Direction.EAST:
-          scaleFactor = yAxis ? 0: 1;
-          break;
-      }
-      if (modifyQueues && (0 !== scaleFactor)) {
-        cmd.opts.queuedDistance -= distThisMove;
+      var moveDirection = utils.normalize(Direction.getUnitVector(cmd.opts.dir));
```
Comment (Hamms) [inline apps/src/studio/studio.js:623]: :+1:

File: apps/test/integration/levelSolutions/studio/k1_6.js [hunk 3]
```diff
@@ -0,0 +1,138 @@
+var testUtils = require('../../../util/testUtils');
```
Comment (Hamms) [inline apps/test/integration/levelSolutions/studio/k1_6.js:1]: why the name `k1_6`?

Comment (Hamms) [inline apps/test/integration/levelSolutions/studio/k1_6.js:1]: ahhh, I see; resurrecting a skipped test. Do we know why that test was marked as skip?

File: apps/src/studio/blocks.js [hunk 4]
```diff
@@ -898,25 +898,50 @@ exports.install = function (blockly, blockInstallOptions) {
...
+        letter: commonMsg.directionSouthLetter(),
         image: skin.downArrow,
-        studioValue: Direction.SOUTH.toString(),
+        studioValue: Direction.SOUTH,
         tooltip: msg.moveDownTooltip()
+      },
+      NorthWest: {
+        letter: commonMsg.directionNorthWestLetter(),
+        image: skin.upLeftArrow,
+        studioValue: Direction.NORTHWEST,
```
Comment (Hamms) [inline apps/src/studio/blocks.js:925]: nit: `NORTH_WEST` would be more consistent

File: apps/src/studio/studio.js [hunk 5]
```diff
@@ -657,6 +646,8 @@ var cancelQueuedMovements = function (index, yAxis) {
         cmd.opts.queuedDistance = 0;
       } else if (!yAxis && (dir === Direction.EAST || dir === Direction.WEST)) {
         cmd.opts.queuedDistance = 0;
+      } else if (!CardinalDirections.includes(dir)) {
```
Comment (Hamms) [inline apps/src/studio/studio.js:649]: `includes` is unsupported in IE; do we have a working babel polyfill?

**Comment by Hamms** [comment]
LGTM after comments!

File: apps/src/studio/blocks.js (hunk 4 above)
Comment (balderdash) [inline apps/src/studio/blocks.js:925]: These constants already existed, and I guess "northwest" is really just one word. I changed the other names to be consistent with this one instead ("NorthWest" => "Northwest", etc.).

File: apps/src/studio/studio.js (hunk 5 above)
Comment (balderdash) [inline apps/src/studio/studio.js:649]: It works for me in IE, I was under the impression that https://github.com/code-dot-org/code-dot-org/pull/8634 gave us access to all the ES6 features in apps and code-studio. Is that not how it works?

File: apps/test/integration/levelSolutions/studio/k1_6.js (hunk 3 above)
Comment (balderdash) [inline apps/test/integration/levelSolutions/studio/k1_6.js:1]: Seems it was disabled for being flaky in https://github.com/code-dot-org/code-dot-org/commit/12429ab046d6083f8b3fff428bccbfd125cd91c8:

> Disable k1_6 test until we can figure out why it randomly fails

I tried running a few dozen times, and at least locally I had no issues with it.

**Pull request merged**